File: features/fintech/tokocash/src/main/java/com/tokopedia/tokocash/ovoactivation/view/IntroOvoActivity.java
package com.tokopedia.tokocash.ovoactivation.view;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;

import com.airbnb.deeplinkdispatch.DeepLink;
import com.tokopedia.tokocash.ApplinkConstant;

/**
 * Created by nabillasabbaha on 20/09/18.
 */
public class IntroOvoActivity extends BaseOvoActivationActivity implements IntroOvoFragment.OvoFragmentListener {

    public static final String TOKOCASH_ACTIVE = "tokocash_active";

    @SuppressWarnings("unused")
    @DeepLink(ApplinkConstant.WALLET_INTRODUCTION_OVO)
    public static Intent getcallingIntent(Context context, Bundle extras) {
        boolean tokocashActive = Boolean.parseBoolean(
                extras.getString(ApplinkConstant.WALLET_TOKOCASH_ACTIVE, "false"));
        return IntroOvoActivity.newInstance(context, tokocashActive);
    }

    public static Intent newInstance(Context context, boolean tokocashActive) {
        Intent intentIntroOvo = new Intent(context, IntroOvoActivity.class);
        intentIntroOvo.putExtra(TOKOCASH_ACTIVE, tokocashActive);
        return intentIntroOvo;
    }

    @Override
    protected Fragment getNewFragment() {
        return IntroOvoFragment.newInstance(getIntent().getBooleanExtra(TOKOCASH_ACTIVE, false));
    }

    @Override
    public void setTitleHeader(String titleHeader) {
        updateTitle(titleHeader);
    }
}

File: features/fintech/wallet/src/main/java/com/tokopedia/wallet/ovoactivation/view/IntroOvoActivity.kt
package com.tokopedia.wallet.ovoactivation.view

import androidx.fragment.app.Fragment

/**
 * Created by nabillasabbaha on 20/09/18.
 */
class IntroOvoActivity : BaseOvoActivationActivity(), IntroOvoFragment.OvoFragmentListener {

    override fun getNewFragment(): Fragment? {
        return IntroOvoFragment.newInstance(isTokoCashActive())
    }

    private fun isTokoCashActive(): Boolean {
        val tokocashActive = intent.data.getQueryParameter(WALLET_PARAM_TOKOCASH_ACTIVE)
        return if (tokocashActive.isNotEmpty())
            tokocashActive.toBoolean()
        else false
    }

    override fun setTitleHeader(titleHeader: String) {
        updateTitle(titleHeader)
    }

    companion object {

        const val TOKOCASH_ACTIVE = "tokocash_active"
        const val WALLET_PARAM_TOKOCASH_ACTIVE = "tokocash"
    }
}
